.github/workflows/python.yml

```diff
@@ -10,19 +10,18 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v4
-    # NB: v1.4.0 covers Python 3.8.
-    - uses: ricardochaves/python-lint@v1.4.0
+    - uses: actions/setup-python@v5
       with:
-        python-root-list: lddtree.py pylint
-        use-pylint: true
-        use-pycodestyle: false
-        use-flake8: false
-        use-black: true
-        use-mypy: true
-        use-isort: true
-        extra-pylint-options: ""
-        extra-pycodestyle-options: ""
-        extra-flake8-options: ""
-        extra-black-options: ""
-        extra-mypy-options: ""
-        extra-isort-options: ""
+        python-version: '3.x'
```

Comment:
if we have control over versions, then we'd want to test a range of versions, starting with 3.6 if possible

Reply:
Lint tools are usually python version agnostic: they'll provide the same style violation reports regardless of which python version the linter tool is installed for.

The linter tools will, in some cases, implement different checks depending on version that you want to be compatible with. For example:

```
mypy --python-version=3.11 lddtree.py
```

will tell mypy to type check against the 3.11 version of the standard library type checking stubs.

```
pylint --py-version=3.5 lddtree.py
```
will tell pylint to not e.g. suggest using an f-string.

Reply:
if that's reliable, that's fine, but it's also something we need in here

Reply:
So to be clear, do you want:

- linters that support checking against multiple python versions, to do so?
  - lower and upper version bounds only? e.g. 3.6 and 3.12?
- linters that support checking against multiple versions, to check against the lowest supported version to avoid suggesting issues that cannot be fixed until bumping the minimum version?

Reply:
for black/isort, we just need to tell them the min version to use as the syntax, so setting it to py3.6 alone is OK

for pylint/mypy, yes, we'd want to check multiple versions.  ideally it'd be 3.6 as the min, but if that's onerous, i'd be OK raising this to py3.8.  but not beyond that atm.

i guess i'd be OK with a min/max version, but seems like if we could do multiple versions, then it'd be easy to list them all inclusive.  i.e. use github actions matrix settings.